bench/workgen/workgen.cxx

```diff
@@ -1360,39 +1385,70 @@ Thread::describe(std::ostream &os) const
 
 Operation::Operation()
     : _optype(OP_NONE), _internal(nullptr), _table(), _key(), _value(), _config(),
-      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0)
+      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0), _random_table(false),
+      _random_kv(false)
 {
     init_internal(nullptr);
 }
 
 Operation::Operation(OpType optype, Table table, Key key, Value value)
     : _optype(optype), _internal(nullptr), _table(table), _key(key), _value(value), _config(),
-      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0)
+      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0), _random_table(false),
+      _random_kv(false)
 {
+    ASSERT(is_table_op());
     init_internal(nullptr);
     size_check();
 }
 
 Operation::Operation(OpType optype, Table table, Key key)
     : _optype(optype), _internal(nullptr), _table(table), _key(key), _value(), _config(),
-      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0)
+      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0), _random_table(false),
+      _random_kv(false)
 {
+    ASSERT(is_table_op());
     init_internal(nullptr);
     size_check();
 }
 
 Operation::Operation(OpType optype, Table table)
     : _optype(optype), _internal(nullptr), _table(table), _key(), _value(), _config(),
-      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0)
+      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0), _random_table(false),
+      _random_kv(false)
 {
+    ASSERT(is_table_op());
     init_internal(nullptr);
     size_check();
 }
 
+Operation::Operation(OpType optype, Key key, Value value)
+    : _optype(optype), _internal(nullptr), _table(), _key(key), _value(value), _config(),
+      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0), _random_table(true),
+      _random_kv(false)
+{
+    ASSERT(is_table_op());
+    init_internal(nullptr);
+}
+
+Operation::Operation(OpType optype)
+    : _optype(optype), _internal(nullptr), _table(), _key(), _value(), _config(),
+      transaction(nullptr), _group(nullptr), _repeatgroup(0), _timed(0.0), _random_table(true),
+      _random_kv(true)
+{
+    ASSERT(is_table_op());
+    init_internal(nullptr);
+    // TODO - Do I need to set this here for now?
```

Comment:
@ddanderson, here we don't have much information, just the operation type.
In order to generate a random key, is it here where I should define if the `keytype` is `PARETO` or `UNIFORM`?

Reply:
Yes, I think here is good.  Would it always be `PARETO`?  (or `UNIFORM`) or does it depend on something at runtime, like which particular table you're using?  That's doable, but a little more involved.